### PR TITLE
Use bytes8 index access

### PR DESCRIPTION
**Pull request by chriseth**
Just came to my mind - not sure if it is more efficient..

### CODE REVIEW

File: deposit_contract.sol [hunk 1]
```diff
@@ -131,13 +131,14 @@ contract DepositContract is IDepositContract {
...
-        ret[0] = bytes1(uint8(value));
-        ret[1] = bytes1(uint8(value >> 8));
-        ret[2] = bytes1(uint8(value >> 16));
-        ret[3] = bytes1(uint8(value >> 24));
-        ret[4] = bytes1(uint8(value >> 32));
-        ret[5] = bytes1(uint8(value >> 40));
-        ret[6] = bytes1(uint8(value >> 48));
-        ret[7] = bytes1(uint8(value >> 56));
+        bytes8 bytesValue = bytes8(value);
+        ret[0] = bytesValue[0];
```
Comment (axic) [inline deposit_contract.sol]: So this works because `uint64` -> `bytes8` casting switches the byteorder and then we are copying it over to `bytes`?

Comment (chriseth) [reply deposit_contract.sol]: Ah sorry, no, the indices have to be reversed!

**Comment by chriseth** [comment]
Without optimizer, this adds range checks and thus is very expensive. With optimizer, it removes the masking and uses `byte` instead of shifts and is cheaper.

**Comment by daejunpark** [comment]
This makes the code a lot simpler and easier to read. I also ran FV on the compiled bytecode of this new code pattern, and it worked well.

**Comment by axic** [comment]
With byteswapping, this passes all the tests now. Let's revert that commit and see if it is detected.

**Comment by axic** [comment]
It does!

**Review by MrChico — APPROVED** [review]

**Pull request merged**